test/e2e/kubectl.go

```diff
@@ -56,49 +48,131 @@ var _ = Describe("kubectl", func() {
-
-		By("creating a replication controller")
-		runKubectl("create", "-f", nautilusPath)
-		validateController(c, nautilusImage, 2)
-		By("scaling down the replication controller")
-		runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=1")
-		validateController(c, nautilusImage, 1)
-		By("scaling up the replication controller")
-		runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=2")
-		validateController(c, nautilusImage, 2)
-	})
+	Describe("guestbook", func() {
+		var (
+			guestbookPath       = filepath.Join(testContext.repoRoot, "examples/guestbook")
+			frontendSelector    = "name=frontend"
+			redisMasterSelector = "name=redis-master"
+			redisSlaveSelector  = "name=redis-slave"
+		)
+
+		It("should create and stop a working application", func() {
+			defer cleanup(guestbookPath, frontendSelector, redisMasterSelector, redisSlaveSelector)
 
-	It("should do a rolling update of a replication controller", func() {
-		// Cleanup all resources in case we fail somewhere in the middle
-		defer cleanup(updateDemoRoot)
+			By("creating all guestbook components")
+			runKubectl("create", "-f", guestbookPath)
 
-		By("creating the initial replication controller")
-		runKubectl("create", "-f", nautilusPath)
-		validateController(c, nautilusImage, 2)
-		By("rollingupdate to new replication controller")
-		runKubectl("rollingupdate", "update-demo-nautilus", "--update-period=1s", "-f", kittenPath)
-		validateController(c, kittenImage, 2)
+			By("validating guestbook app")
+			validateGuestbookApp(c)
+		})
 	})
 
 })
 
-func cleanup(filePath string) {
+func validateGuestbookApp(c *client.Client) {
+	// Wait 2 minutes for frontend to serve content.
+	serving := false
+	for i := 1; i < 24; i++ {
+		entry, err := makeRequestToGuestbook(c, "get", "")
+		if err == nil && entry == "{\"data\": \"\"}" {
+			serving = true
+			break
+		}
+		time.Sleep(5 * time.Second)
+	}
+	if !serving {
+		Failf("Frontend service did not start serving content in 2 minutes")
+	}
+
+	// Try to add a new entry to the guestbook for 2 minutes.
+	added := false
+	for i := 1; i < 24; i++ {
+		result, err := makeRequestToGuestbook(c, "set", "TestEntry")
+		if err == nil && result == "{\"message\": \"Updated\"}" {
+			added = true
+			break
+		}
+		time.Sleep(5 * time.Second)
+	}
+	if !added {
+		Failf("Cannot added new entry in 2 minutes")
+	}
+
+	// Verify that entry is correctly added to the guestbook.
+	entry, err := makeRequestToGuestbook(c, "get", "")
+	if err != nil {
+		Failf("Request to the guestbook failed with: %v", err)
+	}
+	if entry != "{\"data\": \"TestEntry\"}" {
+		Failf("Wrong entry received: %v", entry)
+	}
+}
+
+func makeRequestToGuestbook(c *client.Client, cmd string, value string) (string, error) {
+	result, err := c.Get().
+		Prefix("proxy").
+		Resource("services").
+		Name("frontend").
+		Suffix("/index.php").
+		Param("cmd", cmd).
+		Param("key", "messages").
+		Param("value", value).
+		Do().
+		Raw()
+	return fmt.Sprintf("%s", result), err
```

Comment:
What type is result? I think it's a []byte, in which case you can just return string(result), err here.